pkg/acs/calacs/acsrej/acsrej_sky.c

```diff
@@ -88,52 +108,54 @@ void acsrej_sky (char *sky, IODescPtr ipsci[], IODescPtr ipdq[],
     } /* End of loop over lines */
 
     freeHdr(&dqhdr);
-    
+
     /* Compute min and max for histogram.
-        MIN is min of good data or MINVAL, which ever is greater
-        DELTA is difference between mean of data and MIN
-        MAX is mean plus DELTA
-        This insures that the mean falls in the center of the range
-            between min and max.
-    */
+       MIN is min of good data or MINVAL, which ever is greater
+       DELTA is difference between mean of data and MIN
+       MAX is mean plus DELTA
+       This insures that the mean falls in the center of the range
+       between min and max. */
     if (npt == 0) npt = 1;
     min = (data_min < MINVAL) ? MINVAL : data_min;
     mean = (sum > 0.) ? (int) ( (sum / (float)npt) + 1) : 1;
     max = 2 * mean - min;
-    
+
```

Comment:
This comment is on "old" code which seems to pre-date Pey Lian's updates to the the ACSREJ algorithm.  While the comments in the code on finding the min, mean, and max values are true, this technique ignores values which are actually larger than the pseudo-max.  This is one way to eliminate outliers, but the mean itself has been computed based upon the outliers.  A kappa-sigma clipping is traditionally used to eliminate outliers and find a more robust mean.

Reply:
Note to self: The determination of the mean is probably good enough as the histogram is being used as a filtering mechanism from which a mode value will be determined to characterize an entire input image.

Reply:
Note on note to self: You should note down all these stuff in preparation to write an ACS Instrument Science Report on the new ACSREJ algorithm. :smile:

Reply:
In response to @pllim comment, I was just about to send you an email @mdlpstsci and tell you that I do plan on writing an ISR on the new ACSREJ algorithm because I have been doing a lot of verification of the data products produced. I think it would be a great idea to get together and go over the code in detail, specifically, any changes that were made to the actual thresholding process for determining if a pixel was affected by a cosmic ray. At the present moment, I can say definitively that the ERR arrays are finally correctly handled and match what is expected an error propagation standpoint. I have some nice code to generate region files showing what pixels each implementation finds in common, as well as, what the pixels each finds separately from the other. In order to know if this is expected, I need to understand what code changes where made to the thresholding procedure. I plan on presenting all of this at the next team meeting, so if we could meet sometime before (11/15) that would be best, but meeting after that date is fine.

Reply:
Understood.  I need to stop being distracted and finish my review.  I should make an outline for both of us regarding the new algorithm.